### ISSUE TITLE
Firestore collection counters

### ISSUE THREAD

**Issue by joppehoekstra** [activisthandbook/activisthandbook-editor#1]
**Is your feature request related to a problem? Please describe.**
It's not natively possible to get a count for a Firestore collection without incurring many reads. 

**Describe the solution you'd like**
Keep track of collection items as they are created and deleted and update a counter.

Counters needed:
- Every article has two sub collections*: draftVersions and publishedVersions. We need counters for both, saved in the article document. Should be easily doable using Firestore cloud function triggers on creation and deletion.
- It would be nice to keep track of the number of articles that have a review requested, saved in the 'app/analytics' document. Not quite sure yet how to implement.
- Every languageCollection has two arrays: publishedArticles and draftArticles (right now named 'articles', but I think we should rename this). For both, we want have a count saved in the languageCollection document.

*Right now, these are both in 1 subcollection called 'versions', but I propose splitting them up into 2 subcollections to make counting easier.

**Comment by VladMasarik** [comment]
> 3. Every languageCollection has two arrays

I am not sure what the use case is, but are you certain that we need an extra counter for it? AFAIK if we would like to read the counter from a `languageCollection` we would have to fetch the collection. At that point we have the object loaded locally, and from my understanding the `languageCollection[index].articles_draft` is an array (same for `languageCollection[index].articles_published`), and since we already fetched it, we can just read the array size. Ie. we don't need counter because we have to fetch the collection anyway, and once we do that, we can just read the array length.

However, I have no idea whether this is true, is it the case that we cannot get a specific field only from a document, and not the whole document? Eg. `languageCollections[myIndex].getField(myString)` is this possible? I looked into the docs, and it is not clear to me, but I don't think it is possible.

They have `documentReference` https://firebase.google.com/docs/reference/js/v8/firebase.firestore.DocumentReference , but that seems to be only for fetching the whole document. Once it is fetched, you have `DocumentSnapshot` which has a `get()` method that can read a field, but again, we already fetched the document at that point. There are also `FieldValue` and `FieldPath` objects, but `FieldValue` seems something completely different ("Sentinel value" what is that?), and `FieldPath` has no `get()` method https://firebase.google.com/docs/reference/js/v8/firebase.firestore.FieldValue

If we really do need the counter, one option would be updating the counter right before we update the whole document, however where exactly is this happening? Also, I think this would just add tons of complexity, and we would have to keep in mind to always update the counter before pushing the changes to the DB, or create like a central `languageCollectionManager` which would take care of the updates.

There is one more option, which I read that the Firestore has some listeners https://firebase.google.com/docs/firestore/query-data/listen . Which I think would be kind of an an overkill though, eg. listen for changes on the array, and then update the counter.

I think that if we are forced to fetch the document anyway, then just reading the array length is enough. If you are worried about performance, I think that getting a single document at a time + catching + async loading would be more than enough, since I wouldn't consider the counter value to be a critical information. I could be wrong though, so the whole thing depends on what was your original use case, and whether the array length is not enough.
